### PR TITLE
Made payment page, brandCode and skipSelection overwritable

### PR DESCRIPTION
**Pull request by onchainguy-btc**

### CODE REVIEW

**Comment by hekike** [comment]
@seb0zz looks good but one of the test is broken. I'm not using Adyen a long time ago. Could you help to fix the test as well?

**Comment by onchainguy-btc** [comment]
Yeah, I will take a look at the tests within the following days!

**Comment by hekike** [comment]
@seb0zz thank you!

**Comment by onchainguy-btc** [comment]
Mhhh might it be that the merchant account does not exist anymore or has been changed? The test fails due to the following error:

Error: Merchant signature was not valid!<br />You can test the HMAC Calculation by submitting a payment request to https://ca-test.adyen.com/ca/ca/skin/checkhmac.shtml

**Comment by hekike** [comment]
@seb0zz it's possible. I don't remember where I found that test account. Maybe we should mock it out. What do you think?

**Comment by onchainguy-btc** [comment]
Well in fact I think we don't even need to test the response from adyen but only that the `generateRequest` method is generating a valid URL for adyen. So instead of checking the HTML body, we could just check if the URL contains the `merchantReference` etc. What do you think?

**Comment by hekike** [comment]
@seb0zz yes but the idea was behind that checking the validity of the generated request. But let's to this, better then no running test yet :D

**Comment by onchainguy-btc** [comment]
Sorry for the delay, I think I can do it tomorrow!

**Comment by onchainguy-btc** [comment]
Alright, what do you think? Merge? :-)

File: test/hpp.js [hunk 1]
```diff
@@ -37,23 +36,18 @@ describe('HPP', function () {
 
     it('should generate url and merchantReference', function (done) {
       hppPayment.generateRequest(function (err, url, merchantReference) {
+        console.log(url);
```
Comment (hekike) [inline test/hpp.js]: please remove console.log

**Comment by hekike** [comment]
looks good, only a small comment ;)

**Comment by onchainguy-btc** [comment]
Oh no I'm sorry :eyes: removed the log output!

**Comment by hekike** [comment]
v1.0.0 rolled out. Yes it wasn't proper SemVer :P my fault...
https://www.npmjs.com/package/adyen ;)

**Comment by hekike** [comment]
Thanks for the collaboration!

**Comment by onchainguy-btc** [comment]
Thanks man! :heart_eyes:

**Comment by hekike** [comment]
Oh one more thing, could you put to the README in an other PR as well? I'm pretty sure that would be helpful for the others.

**Pull request merged**